### PR TITLE
[ReadyforReview]Iss287 add seq from array class method to input

### LINKED ISSUE #287: Add `seq_from_array` class method to `Input`
## Motivation

It is likely common for users to have a collection of simulator inputs as a 2-d Numpy array, where each row corresponds to a single simulator input. It would therefore be good to create a class method of `Input` that creates a tuple of `Input`s from such an array.

## Acceptance criteria

Implement such a class method, taking care to do any required argument validation and to write unit tests.

### PR DESCRIPTION
**Pull request by HarryWhiteRSE**
Closes #287 

Create a new `Input` class method: `sequence_from_array`. This allows the user to create a `tuple` of `Input` points from either a 2D array or sequence of 1D arrays. 

Validated the argument entry to `sequence_from_array` for `Sequence` and `NDarray` objects however, all internal validation falls within `from_array`. (ie. if you pass a sequence of integers rather than arrays, `from_array` will catch these).  

Also contains a set of unit tests to ensure that the function works as intended.

### CODE REVIEW

File: tests/unit/test_modelling.py [hunk 1]
```diff
@@ -278,6 +278,85 @@ def test_from_array_not_finite_error(self):
...
+
+        # 1D case
+        _inputs = np.array([1, 2, 3])
+        with self.assertRaisesRegex(ValueError, exact(msg)):
+            _ = Input.sequence_from_array(_inputs)
+
+        # Empty array case
+        _inputs = np.array([])
+        with self.assertRaisesRegex(ValueError, exact(msg)):
+            _ = Input.sequence_from_array(_inputs)
```
Comment (JSalter90) [inline tests/unit/test_modelling.py:339]: Need to redefine dimensions of `_inputs`.

```suggestion
        # 1D case
        _inputs = np.array([1, 2, 3])
        msg = f"Expected np.array of dimension 2, but received {_inputs.ndim} dimensions."
        with self.assertRaisesRegex(ValueError, exact(msg)):
            _ = Input.sequence_from_array(_inputs)

        # Empty array case
        _inputs = np.array([])
        msg = f"Expected np.array of dimension 2, but received {_inputs.ndim} dimensions."
        with self.assertRaisesRegex(ValueError, exact(msg)):
            _ = Input.sequence_from_array(_inputs)
```

**Review by JSalter90 — CHANGES_REQUESTED** [review]
Suggested small fix so tests work

File: tests/unit/test_modelling.py (hunk 1 above)
Comment (mbjohns) [reply tests/unit/test_modelling.py:339]: Ah, you spotted my deliberate mistake!

**Review by JSalter90 — APPROVED** [review]
LGTM 🚀

**Review by mbjohns — APPROVED** [review]
LGTM 🚀

**Pull request merged**